Add x402

sriakula1 opened the pull request:
**https://github.com/xpaysh/awesome-x402#readme**

**Why it should be included:** x402 is the HTTP 402 Payment Required protocol - a web standard for internet-native payments. It enables AI agents to autonomously pay for API access and services. The protocol is backed by Coinbase and Cloudflare, with 500K+ weekly transactions.

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

## Requirements for your pull request

- [x] Don't open a Draft / WIP pull request while you work on the guidelines. A pull request should be 100% ready and should adhere to all the guidelines when you open it.
- [x] **Don't waste my time.** Do a good job, adhere to all the guidelines, and be responsive.
- [x] **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).**
    - Reviewed #3781 (Add Parquet) - Found inconsistent punctuation issues
    - Reviewed #3737 (Add PDF) - Suggested description improvement
    - Reviewed #3719 (Add Vibe Coding) - Found missing periods in descriptions, noted format variations
    - Reviewed #3713 (Add Product Engineer) - Well-structured list, minor description suggestion
- [x] You have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/main/create-list.md).
- [x] This pull request has a title in the format `Add Name of List`. It should not contain the word `Awesome`.
- [x] Your entry here should include a short description of the project/theme of the list. **It should not describe the list itself.** The first character should be uppercase and the description should end in a dot.
- [x] Your entry should be added at the bottom of the appropriate category.
- [x] The title of your entry should be title-cased and the URL to your list should end in `#readme`.
- [x] No blockchain-related lists.
- [x] The suggested Awesome list complies with the below requirements.

## Requirements for your Awesome list

- [x] **Has been around for at least 30 days.** First commit: 2025-11-02 (36+ days)
- [x] Run [`awesome-lint`](https://github.com/sindresorhus/awesome-lint) on your list and fix the reported issues.
- [x] The default branch should be named `main`, not `master`.
- [x] **Includes a succinct description of the project/theme at the top of the readme.**
- [x] It's the result of hard work and the best I could possibly produce.
- [x] The repo name of your list should be in lowercase slug format: `awesome-x402`.
- [x] The heading title of your list should be in title case format: `# Awesome X402`.
- [x] Non-generated Markdown file in a GitHub repo.
- [x] The repo should have `awesome-list` & `awesome` as GitHub topics.
- [x] Not a duplicate. Please search for existing submissions.
- [x] Only has awesome items. Awesome lists are curations of the best, not everything.
- [x] Does not contain items that are unmaintained, has archived repo, deprecated, or missing docs.
- [x] Includes a project logo/illustration whenever possible.
- [x] Entries have a description, unless the title is descriptive enough by itself.
- [x] Includes the Awesome badge.
- [x] Has a Table of Contents section.
- [x] Has an appropriate license: CC0-1.0.
- [x] Has contribution guidelines.
- [x] All non-important but necessary content grouped in a Footnotes section at the bottom of the readme.
- [x] Has consistent formatting and proper spelling/grammar.
- [x] Does not use hard-wrapping.
- [x] Does not include a CI badge.
- [x] Does not include an `Inspired by awesome-foo` or `Inspired by the Awesome project` link at the top of the readme.

---

unicorn

sriakula1 committed:
Add x402

Mo-way reviewed (changes requested):
Sorry to say this:
In my opinion, you didn't do an actual review of other PRs. Your "reviews" are AI generated and basically a long "lgtme", but in form that takes 30s instead of 1s to parse mentally.

MalloryWittwer reviewed (changes requested):
Hey @sriakula1 , thank you for submitting your list. After reviewing it, I have a few suggestions on how you could improve it:

- The title of your list is "Awesome X402". I suppose the title of your PR should therefore be "Add X402" with a capitalized X, for consistency?
- In your entry, the "x" in "x402" should probably be capitalized to comply with the PR template requirements.
- The short description next to your entry could probably be shorter; I would suggest simply "HTTP 402 Payment Required protocol".
- Your entry is at the bottom of the "Decentralized Systems" section. Would it not make more sense to have it at the end of "Networking" (I don't have a strong opinion on this)?
- I find that how you present your list at the top of the README as "The Ultimate x402 Resource Hub..." sounds a bit like marketing blurb. Could you consider a simple and objective description of what X402 is and its interest instead? For the same reason, I would suggest removing the lines around "Start building in 5 minutes [...]" below the Contents section.
- I see that Claude is a contributor to the project. This makes me wonder if and how much you've used AI to help you write your list?
- I would suggest removing the extended parts of your list that don't contain links, for example those under the "Production Implementations" or "Use Cases & Patterns" section.
- I would suggest to keep only a link to your CONTRIBUTING.md file under the "Contributing" section at the end of your list, and to remove the rest of that section, so you don't have to maintain the contributing guidelines in two places.

Looking forward to see the final version of your list!

sriakula1 committed:
Update entry based on PR feedback

- Capitalize X402 in title for consistency
- Shorten description to focus on HTTP protocol

sriakula1 commented on readme.md:
```diff
@@ -731,6 +731,7 @@
 - [Cosmos SDK](https://github.com/cosmos/awesome-cosmos#readme) - Modular framework for building app-specific blockchains in Go.
 - [Tor](https://github.com/polycarbohydrate/awesome-tor#readme) - A free overlay network for enabling anonymous communication.
 - [ATProto](https://github.com/atblueprints/awesome-atproto#readme) - Open, decentralized network for building social apps.
+- [x402](https://github.com/xpaysh/awesome-x402#readme) - HTTP 402 Payment Required protocol for internet-native payments and AI agent commerce.
```
it's an HTTP protocol (uses 402 status code) that happens to be championed by Coinbase at the moment, along with Cloudflare, VISA, MasterCard etc. Similar to how a payments API might use Stripe on the backend but isn't a "Stripe list". But I get the concern, happy to discuss with maintainers if needed, as it's gonna be the internet payment rails and while some payments may happen on-chain, there's a significant non-blockchain element to it.

sriakula1 commented:
> Sorry to say this: In my opinion, you didn't do an actual review of other PRs. Your "reviews" are AI generated and basically a long "lgtme", but in form that takes 30s instead of 1s to parse mentally.

Did use AI but not entirey. However, noted. Will be more specific with reviews.

Mo-way reviewed (changes requested):
IMO: You repo reads like a marketing post, not an awesome list. 
See feedback by [MalloryWittwer](https://github.com/MalloryWittwer).

Also, if you use claude to do the work, I think it is _extremely_ rude to just add its content to PRs  and let others check for issues like broken links.

Mo-way commented:
> > Reviewed this list. Found multiple broken links in Protocol Implementations (all "Other Languages" entries), Example Applications, Tools, Testing, and Tutorials sections. These return 404 errors.
> 
> Legit feedback, broken links are now fixed. Thx for the review

Oh wow, the topic actually sparked my interest. But it turns out, the [commit](https://github.com/xpaysh/awesome-x402/commit/971fba644c1185596485d8202485f9a73ff96e00
) that "replaces the dead links" removes a lot of dead links, while leaving the remaining dead links unchanged...

So much for your claim "_Did use AI but not entirey_"... I'm lost for words, so I asked ChatGPT how to reply this behavior:

“Your review process is like a unicorn—everyone talks about it, nobody’s seen it.”

sindresorhus commented:
> No blockchain-related lists.
